src/wp-includes/rest-api/endpoints/class-wp-rest-global-styles-controller.php

```diff
@@ -10,24 +10,20 @@
 /**
  * Base Global Styles REST API Controller.
  */
-class WP_REST_Global_Styles_Controller extends WP_REST_Controller {
-
-	/**
-	 * Post type.
-	 *
-	 * @since 5.9.0
-	 * @var string
-	 */
-	protected $post_type;
-
+class WP_REST_Global_Styles_Controller extends WP_REST_Posts_Controller {
 	/**
 	 * Constructor.
+	 *
 	 * @since 5.9.0
+	 * @since 6.5.0 Extends class from WP_REST_Posts_Controller.
+	 *
+	 * @param string $post_type Post type.
 	 */
-	public function __construct() {
-		$this->namespace = 'wp/v2';
-		$this->rest_base = 'global-styles';
-		$this->post_type = 'wp_global_styles';
+	public function __construct( $post_type ) {
+		$this->post_type = $post_type;
+		$obj             = get_post_type_object( $post_type );
+		$this->rest_base = ! empty( $obj->rest_base ) ? $obj->rest_base : $obj->name;
+		$this->namespace = ! empty( $obj->rest_namespace ) ? $obj->rest_namespace : 'wp/v2';
 	}
```

Comment:
I don't think we need to override the __construct here. I think the original one will work just fine.

Reply:
I think I left it out because `WP_REST_Posts_Controller` adds meta to the class object, `$this->meta = new WP_REST_Post_Meta_Fields( $this->post_type );`, but as you say, it works fine and as expected without the constructor. 

I'll remove. Cheers!